src/data/list/range.lean

```diff
@@ -301,8 +301,21 @@ theorem of_fn_eq_map {α n} {f : fin n → α} :
   of_fn f = (fin_range n).map f :=
 by rw [← of_fn_id, map_of_fn, function.right_id]
 
-theorem nodup_of_fn {α n} {f : fin n → α} (hf : function.injective f) :
+theorem nodup_of_fn_of_injective {α n} {f : fin n → α} (hf : function.injective f) :
```

Comment:
It's unfortunate that `of_fn` has a name like that, it makes all the names involving it hard to read. I'm thinking perhaps dot notation might be a nice way around it:
```suggestion
theorem _root_.function.injective.nodup_of_fn {α n} {f : fin n → α} (hf : function.injective f) :
```

Reply:
I don't really like moving this out of the `list` namespace, because neither `nodup` or `of_fn` are immediately  recognizable as list operators.

Reply:
How about (in a future PR) renaming `list.of_fn` to `fin.to_list` or `function.to_list` or `tuple.to_list`?